contrib/hawq-docker/centos7-docker/hawq-test/service-pxf.sh

```diff
@@ -0,0 +1,226 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+build() {
+    source ${HAWQ_HOME}/greenplum_path.sh
+    export PXF_HOME=${GPHOME}/pxf
+
+    sudo echo "source ${GPHOME}/greenplum_path.sh" >> /home/gpadmin/.bashrc
+
+    cd /data/hawq/pxf
+    make -j8
+    make install
+
+    sudo sed 's|-pxf|-gpadmin|g' -i ${PXF_HOME}/conf/pxf-env.sh
+
+    rm -rf ${PXF_HOME}/conf/pxf-log4j.properties
+    cat <<EOF >>${PXF_HOME}/conf/pxf-log4j.properties
+    # Licensed to the Apache Software Foundation (ASF) under one
+    # or more contributor license agreements.  See the NOTICE file
+    # distributed with this work for additional information
+    # regarding copyright ownership.  The ASF licenses this file
+    # to you under the Apache License, Version 2.0 (the
+    # "License"); you may not use this file except in compliance
+    # with the License.  You may obtain a copy of the License at
+    #
+    #   http://www.apache.org/licenses/LICENSE-2.0
+    #
+    # Unless required by applicable law or agreed to in writing,
+    # software distributed under the License is distributed on an
+    # "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    # KIND, either express or implied.  See the License for the
+    # specific language governing permissions and limitations
+    # under the License.
+    log4j.appender.ROLLINGFILE.File=${PXF_HOME}/pxf-service/logs/pxf-service.log
+    log4j.appender.ROLLINGFILE.MaxFileSize=10MB
+    log4j.appender.ROLLINGFILE.MaxBackupIndex=10
+    log4j.appender.ROLLINGFILE.layout=org.apache.log4j.PatternLayout
+    log4j.appender.ROLLINGFILE.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss.SSSS} %p %t %c - %m%n
+EOF
+
+    rm -rf ${PXF_HOME}/conf/pxf-private.classpath
+    cat <<EOF >>${PXF_HOME}/conf/pxf-private.classpath
```

Comment:
If https://github.com/apache/hawq/blob/master/pxf/pxf-service/src/configs/templates/pxf-private-hdp.classpath.template changes, someone who does that may forget to make the same changes in the current script.

I wonder whether it is really necessary to replace `${PXF_HOME}/conf/pxf-private.classpath` with the configuration below. If so, I would propose to generate classpath from the template (right here).

Reply:
@leskin-in I think probably we should use some inspiration from #1335. Have a make `variable` with the Hadoop distro (`cloudera`, `hdp`, `emr`, etc) and load the template during runtime. That way we only have one centralized place with all the classpath templates and each process can load them however they want.

>I wonder whether it is really necessary to replace ${PXF_HOME}/conf/pxf-private.classpath with the configuration below. 
Unfortunately, this is needed :(

Reply:
Good, I think [this code](https://github.com/apache/hawq/pull/1335/files#diff-9d2f9c6bc83265203f79059b3559ce51R345) does the classpath generation mentioned. :+1: for your solution.

Reply:
Fixed